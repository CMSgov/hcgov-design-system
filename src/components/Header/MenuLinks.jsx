import PropTypes from 'prop-types';
import React from 'react';

/**
 * The primary list of menu links. This ensures a consistent styling
 * across header variations. On the homepage, this list is displayed
 * horizontally on desktop. In other versions, this list is visible
 * when the "Menu" button is toggled and displayed as a stacked list.
 */
const MenuLinks = function (props) {
  return (
    <ul className="hc-c-menu__links ds-c-list ds-c-list--bare">
      {props.links.map(function (link) {
        return (
          <li key={link.href} className="ds-u-margin--0">
            <a
              href={link.href}
              onClick={link.onClick}
              className="hc-c-menu__link"
            >
              {link.label}
            </a>
          </li>
        );
      })}
    </ul>
  );
};

MenuLinks.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      href: PropTypes.string.isRequired,
      label: PropTypes.node.isRequired,
      onClick: PropTypes.func,
    })
  ).isRequired,
};

export default MenuLinks;
