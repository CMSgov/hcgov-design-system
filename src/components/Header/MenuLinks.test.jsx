import MenuLinks from './MenuLinks';
import React from 'react';
import { shallow } from 'enzyme';

describe('MenuLinks', function () {
  it('renders list of links', () => {
    expect(
      shallow(
        <MenuLinks
          links={[
            { href: '#foo', label: 'Foo' },
            { href: '#bar', label: 'Bar' },
            { href: '#baz', label: 'Baz', onClick: () => {} },
          ]}
        />
      )
    ).toMatchSnapshot();
  });
});
