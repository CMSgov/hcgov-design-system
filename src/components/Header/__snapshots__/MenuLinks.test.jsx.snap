// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLinks renders list of links 1`] = `
<ul
  className="hc-c-menu__links ds-c-list ds-c-list--bare"
>
  <li
    className="ds-u-margin--0 "
    key="#foo"
  >
    <a
      className="hc-c-menu__link"
      href="#foo"
    >
      Foo
    </a>
  </li>
  <li
    className="ds-u-margin--0 "
    key="#bar"
  >
    <a
      className="hc-c-menu__link"
      href="#bar"
    >
      Bar
    </a>
  </li>
  <li
    className="ds-u-margin--0 "
    key="#baz"
  >
    <a
      className="hc-c-menu__link"
      href="#baz"
      onClick={[Function]}
    >
      Baz
    </a>
  </li>
</ul>
`;
